pyxem/generators/red_intensity_generator.py

```diff
@@ -0,0 +1,213 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Reduced intensity generator and associated tools.
+
+
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+
+from hyperspy.signals import Signal1D
+
+from pyxem.signals.electron_diffraction1d import ElectronDiffraction1D
+from pyxem.signals.reduced_intensity_profile import ReducedIntensityProfile
+
+from pyxem.components.scattering_fit_component import ScatteringFitComponent
+from pyxem.utils.ri_utils import scattering_to_signal
+
+
+class ReducedIntensityGenerator():
+    """Generates a reduced intensity profile for a specified diffraction radial
+    profile.
+
+
+    Parameters
+    ----------
+    signal : ElectronDiffraction1D
+        An electron diffraction radial profile.
+    """
+
+    def __init__(self, signal, *args, **kwargs):
+        self.signal = signal
+        self.cutoff = [0, signal.axes_manager.signal_axes[0].size - 1]
+        self.nav_size = [signal.axes_manager.navigation_axes[0].size,
+                         signal.axes_manager.navigation_axes[1].size]
+        self.sig_size = [signal.axes_manager.signal_axes[0].size]
+        self.background_fit = None  # added in one of the fits below.
+        self.normalisation = None
+
+    def specify_scattering_calibration(self, calibration):
+        """
+        Defines calibration for the signal axis variable s in terms of
+        A^-1 per pixel. Note that s is defined here as
+        s = 2 sin(theta)/lambda = 1/d.
+
+        Parameters
+        ----------
+        calibration: float
+                    Calibration in terms of A^-1 per pixel.
+        """
+        self.signal.axes_manager.signal_axes[0].scale = calibration
+        return
+
+    def specify_cutoff_vector(self, s_min, s_max):
+        """
+        Scattering vector cutoff for the purposes of fitting an atomic scattering
+        factor to the 1D profile. Specified in terms of s (in inverse angstroms).
+        s is defined as s = 2 sin(theta)/lambda = 1/d.
+
+        Parameters
+        ----------
+        s_min: float
+                    Minimum scattering vector amplitude for cutoff.
+        s_max: float
+                    Maximum scattering vector amplitude for cutoff.
+        """
+        #s_scale = self.signal.axes_manager.signal_axes[0].scale
+        self.cutoff = [s_min, s_max]
+        return
+
+    def fit_atomic_scattering(self, elements, fracs,
+                              N=1., C=0., type='lobato',
+                              plot_fit=True):
+        """Fits a diffraction intensity profile to the background using
+        FIT = N * sum(ci * (fi^2) + C)
+
+        The cutoff for the scattering factor fit to s is defined via the function
+        specify_cutoff_vector above.
+
+        Parameters
+        ----------
+        elements: list of str
+                    A list of elements present (by symbol).
```

Comment:
Can you give an example here, to be explicit, please

Reply:
Is it necessary to observe any special order here? It's best if not, but sometimes that slips in by mistake.

Reply:
No order necessary. Will mention.